### ISSUE TITLE
Named file descriptors

### ISSUE THREAD

**Issue by ZoogieZork** [koalaman/shellcheck#413]
Shellcheck 0.3.8 doesn't seem to recognize "named" file descriptors in redirections in Bash.

For example:

```bash
exec 4>&1  # This is accepted.
exec {foo}>&2  # This isn't.

echo 'test1' >&4
echo 'test2' >&"$foo"  # Ditto.

exec 4>&-
exec {foo}>&-
```

Running this through `shellcheck -s bash redir.sh`:

```
In redir.sh line 4:
exec {foo}>&2
^-- SC2093: Remove "exec " if script should continue after this command.


In redir.sh line 7:
echo "$foo"
      ^-- SC2154: foo is referenced but not assigned.
```

To be clear, the Bash feature I'm referring to is [this one](http://www.gnu.org/software/bash/manual/html_node/Redirections.html):

> Each redirection that may be preceded by a file descriptor number may instead be preceded by a word of the form {varname}. In this case, for each redirection operator except >&- and <&-, the shell will allocate a file descriptor greater than 10 and assign it to {varname}. If >&- or <&- is preceded by {varname}, the value of varname defines the file descriptor to close.

**Comment by justinforce** [comment]
This just bit me, too. I'm doing exactly the same thing.

**Comment by koalaman** [comment]
Fixed in 0f00de8
